aten/src/ATen/SparseTensorImpl.h

```diff
@@ -184,8 +184,11 @@ struct CAFFE2_API SparseTensorImpl : public TensorImpl {
   // make it happen
   void set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values);
 
-  // NOTE: `shallow_copy_and_detach()` does not copy the AutogradMeta pointer
-  // because it is unique for each Variable.
+  // NOTE: `shallow_copy_and_detach()` does not copy the following TensorImpl fields:
+  // 1. the AutogradMeta pointer, because it is unique for each Variable.
+  // 2. the version counter, because we should decide whether the new TensorImpl should share version
+  // counter with the original TensorImpl case-by-case. See NOTE [ Version Counter Sharing ] for details.
```

Comment:
Err, really? I thought if you do a shallow copy, the version counter should always come along for the ride, because the storage is being shared. Looking at the note, I agree that `set_data` should not share, but that seems like the odd case out of all the rest.